### DISCUSSION TITLE
Handling Deep Links

### DISCUSSION THREAD

**Discussion by Kirk-iOS** [dotnet/maui#17863] [Q&A]
What is the correct way to handle deep links for Android? There seems to be a lot of information out there that is no longer relevant. I guess this has topic has been evolving quite a bit over the last few years.

Currently on Android I have created a IntentFilter and it does open my app. I have some background tasks to perform before navigating the user to the correct page. I read about the OnAppLinkRequestReceived method but this never gets called. I seem to think it is linked to the AppLink class and I need to register these links first. It looks like this used to be done on a Navigation page but then moved to the AppShell but I don't see it there either. 

To summarize:

How do I use OnAppLinkRequestReceived to handle my deep link?

Thanks.
